lib/utils.pm

```diff
@@ -1658,6 +1659,43 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check) = splice(@_, 0, 2);
+    my %args = testapi::compat_args({retry => 10, delay => 30, timeout => 90}, [], @_);
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+
+    my $exec = "timeout " . ($args{timeout} - 3) . " $cmd";
```

Comment:
with timeout approach, you also need to care about `!` like https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1605

You are doing the same as `script_retry()` but `script_retry()` is adding the timeout to the final function, while you are reducing the time of the command to be executed, see: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1613

Reply:
ok, I'll add that check too.